scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpression.scala

```diff
@@ -80,6 +81,17 @@ class SpelExpressionEvaluationException(val expression: String, val ctxId: Strin
       cause = cause
     )
 
+sealed trait SpelTemplateSubexpression
+
+object SpelTemplateSubexpression {
+  final case class NonTemplatedValue(val value: String) extends SpelTemplateSubexpression
+
+  final case class TemplatedExpression(expression: SpelExpression) extends SpelTemplateSubexpression {
```

Comment:
We should figure out other naming for this distinction. This names should be usable in human spoken sentences such as "Expression called spel template is a an expression that consists of parts. Each part can be either ... or  ... ". Other example "Spel template expression is a string in which you can use ..."

Currenlty, you will have something like templated expression is an expression that consist of templated expressions and non-templated expressions.

Helper words: placeholder, literal, evalable part, ...

Reply:
You're right, changed it:
1. On level of SpelExpression (scenario-compiler) its: SpelTemplateExpressionPart thats either Literal or Placeholder
2. On level of LazyParameter (api) its: TemplateExpressionPart thats either Literal or Placeholder.